app/src/main/java/de/westnordost/streetcomplete/data/osm/edits/create/CreateNodeFromVertexAction.kt

```diff
@@ -0,0 +1,82 @@
+package de.westnordost.streetcomplete.data.osm.edits.create
+
+import de.westnordost.streetcomplete.data.osm.edits.ElementEditAction
+import de.westnordost.streetcomplete.data.osm.edits.ElementIdProvider
+import de.westnordost.streetcomplete.data.osm.edits.IsActionRevertable
+import de.westnordost.streetcomplete.data.osm.edits.NewElementsCount
+import de.westnordost.streetcomplete.data.osm.edits.update_tags.RevertUpdateElementTagsAction
+import de.westnordost.streetcomplete.data.osm.edits.update_tags.StringMapChanges
+import de.westnordost.streetcomplete.data.osm.edits.update_tags.changesApplied
+import de.westnordost.streetcomplete.data.osm.mapdata.ElementKey
+import de.westnordost.streetcomplete.data.osm.mapdata.ElementType
+import de.westnordost.streetcomplete.data.osm.mapdata.MapDataChanges
+import de.westnordost.streetcomplete.data.osm.mapdata.MapDataRepository
+import de.westnordost.streetcomplete.data.osm.mapdata.Node
+import de.westnordost.streetcomplete.data.osm.mapdata.key
+import de.westnordost.streetcomplete.data.upload.ConflictException
+import de.westnordost.streetcomplete.util.ktx.containsExactlyInAnyOrder
+import kotlinx.serialization.Serializable
+
+/** Action that transforms a way vertex into a node, i.e. adds tags to a vertex in a way.
+ *
+ *  A node in a way (a vertex) without any tags has no meaning in itself but just serves to define
+ *  the geometry of a way. Hence, if we add tags to such a node, it is like creating a new node -
+ *  not on the technical level, but on the semantic level.
+ *
+ *  So, when this is edit is applied, it should be made sure that the vertex didn't change its
+ *  meaning in the definition of the geometry: I.e. its position and the ways it is part of.
+ *
+ *  So, a conflict will be thrown if:
```

Comment:
Probably it should also count as a confict if the vertex now has tags, because "A node in a way (a vertex) without any tags has no meaning in itself but just serves to define the geometry of a way" isn't correct any more.

Reply:
Hm well, the action exists because the assumption is that the node did not have any intrinsic meaning other than for geometry before. But there is no reason why it couldn't also have had an intrinsic meaning before. The important part is the part about the geometry.

For example, a node that is also a `highway=crossing`, like in the screenshot, actually. Or an entrance of a building.

Reply:
If one user adds an entrance to a building node, and meanwhile someone else has added `amenity=post_box` to the same node, should adding `entrance=yes` to the post box really be uploaded?

Reply:
Well, nevermind someone adding `amenity=post_box` in the meantime - what should happen if it already **has** (such) tags? After all, being able to add `traffic_calming=choker` to a crossing or an address to a `entrance=yes` is certainly something that should be possible. And `CreateNodeFromVertexAction` is the right action to choose because the node is selected due to its geometry, not due to it having certain tags.

Reply:
Then the node had tags before, and

> A node in a way (a vertex) without any tags has no meaning in itself but just serves to define  the geometry of a way. Hence, if we add tags to such a node, it is like creating a new node - not on the technical level, but on the semantic level.

is not correct, because it's not like creating a new node

Reply:
I understand your point. The sentence you cite refers to a vertex without tags.  But a vertex *with* tags still retains the property that its meaning is **also** to define geometry of the way. Anyway, I'll amend the doc comment to mention that.